libc/resolve_path.c

```diff
@@ -475,6 +475,48 @@ TEST(resolve_path, symlink_rename)
 }
 
 
+TEST(resolve_path, symlink_long_resolution)
+{
+	const char *dev = "/dev";
+	const char *pathSegment = "dev/../";
+	const char symNameShort[] = "symShort";
+	char symNameLong[NAME_MAX / 2];
+	char basePath[PATH_MAX - 5] = "symShort/";
+	char resolved[PATH_MAX + 1];
+	int basePathLen = strlen(basePath);
+	const int pathSegmentLen = strlen(pathSegment);
+
+	while (basePathLen + pathSegmentLen < PATH_MAX - 10) {
+		strcat(basePath, pathSegment);
+		basePathLen += pathSegmentLen;
+	}
+	strcat(basePath, "dev");
+
+	memset(symNameLong, 'a', NAME_MAX / 2);
```

Comment:
use `sizeof(symNameLong)`  and similar instead of direct values (that way eventual future len change would need to be done only in one place)